nmap_dns.cc

```diff
@@ -807,6 +819,11 @@ static void read_evt_handler(nsock_pool nsp, nsock_event evt, void *) {
       }
     }
   }
+
+  if (!processing_successful){
+    sockaddr_storage discard;
+    process_result(discard, "", ACTION_SYSTEM_RESOLVE, p.id);
```

Comment:
I don't think we want to system resolve *everything* that receives a response without an answer in it. I think instead we should be checking for truncated packets here (and not above). If the packet was still not processed (i.e. not truncated, not cname, no matching PTR record), then we output a debugging/trace statement and just return.